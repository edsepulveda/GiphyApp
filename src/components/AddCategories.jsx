import { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategories = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('')
  const onInputChange = ({ target }) => {
    setInputValue(target.value)
  }

  const onSubmit = (event) => {
    event.preventDefault()
    const newInputValue = inputValue.trim()
    if (newInputValue.length <= 1) return
    setInputValue('')
    onNewCategory(newInputValue)
  }

  return (
    <form onSubmit={onSubmit} aria-label='form'>
      <input
        type='text'
        placeholder='Ingrese nombre de Gifs...'
        value={inputValue} onChange={onInputChange}
      />
    </form>
  )
}

AddCategories.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}
