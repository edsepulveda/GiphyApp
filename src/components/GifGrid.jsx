import GifItem from "./GifItem"
import { useFetchGifs } from "../hooks/useFetchGifs"

export const GifGrid = ({ category }) => {

  const { images, isLoading } = useFetchGifs( category )

  return (

    <>
    <h1>{ category }</h1>
    {
      isLoading && (<h2>Loading...</h2>)
    }
    <div className="card-grid">
      {
        images.map( ( image ) =>(
          <GifItem key={ image.id }
          //Modulo Spread hacia imagen para recibir todas las props que especificamos en GifItem [title,url,id] 
          { ...image }
          
          />
        ) )
      }
    </div>
    </>
  )
}
