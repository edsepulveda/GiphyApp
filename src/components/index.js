// Archivo de barril para nuestro proyecto
// Nos permite un codigo m√°s fluido y entendible para los diferentes usuarios que ingresen al Source Code.

export * from './AddCategories'
export * from './GifGrid'
export * from './GifItem'
