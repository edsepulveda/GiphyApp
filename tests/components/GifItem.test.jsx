import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem'

describe('Evaluacion de proptypes', () => {
  const title = 'Goku'
  const url = 'https://readthedocs.io/'

  test('Should pass', () => {
    const { container } = render(<GifItem title={title} url={url} />)
    expect(container).toMatchSnapshot()
  })

  test('Los atributos de la imagen deben estar validados correctamente', () => {
    render(<GifItem title={title} url={url} />)
    // screen.debug()
    // expect( screen.getByRole('img').src).toBe( url )
    // expect( screen.getByRole('img').alt).toBe( title )
    const { alt, src } = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })
  test('Debe mostrar el texto correspondiente', () => {
    render(<GifItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy()
  })
})
