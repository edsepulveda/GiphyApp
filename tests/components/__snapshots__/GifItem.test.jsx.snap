// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Evaluacion de proptypes Should pass 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://readthedocs.io/"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
